crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1886,18 +1886,27 @@ mod tests {
     const MAC_EOL: &str = "\r";
     const UNIX_EOL: &str = "\n";
 
+    /// Same as [`Token`] except that this includes the [`TokenValue`] as well.
     struct TestToken {
         kind: TokenKind,
         value: TokenValue,
         range: TextRange,
+        flags: TokenFlags,
     }
 
     impl std::fmt::Debug for TestToken {
         fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-            if matches!(self.value, TokenValue::None) {
-                write!(f, "{:#?}", (self.kind, self.range))
+            let mut tuple = f.debug_tuple("");
+            let mut tuple = if matches!(self.value, TokenValue::None) {
+                tuple.field(&self.kind)
+            } else {
+                tuple.field(&self.value)
+            };
+            tuple = tuple.field(&self.range);
```

Comment:
Nit: At least debug_map etc. mutate the state in place. 
```suggestion
						if matches!(self.value, TokenValue::None) {
                tuple.field(&self.kind);
            } else {
                tuple.field(&self.value);
            };
						tuple.field(&self.range);
```

Reply:
Are you suggesting to use `debug_map`? I want to keep the tuple here to have minimal diff which makes it easier to verify the changes. We can explore other formats as a follow-up.

Reply:
Oh no, I'm suggesting to not use `let mut tuple = tuple.field` because I think it might not be necessary (it's not necessary for `debug_map`)